src/pybind/mgr/dashboard/tools.py

```diff
@@ -42,15 +49,22 @@ def _setup(self):
     def request_begin(self):
         req = cherrypy.request
         user = JwtManager.get_username()
+        # In PY2, because `user` is unicode type, `req.path_info` (str) is
```

Comment:
User should not be unicode type, if you're unsure, use `six.ensure_str`.

Reply:
Thanks for the feedback. Originally I was following the sandwich method in this [guide](https://nedbatchelder.com/text/unipain.html), decoding as soon as possible, manipulating Unicode strings, and encoding to bytes again in the last stage.

Turn out it's will be much cleaner and lesser code to just encoding the Unicode string to str type first. I have updated the code and ported the `ensure_str` function since most downstream distributions are still using older versions of six.

The old implementation is kept in https://github.com/bk201/ceph/commit/d001b2cdfd4b283a5904d6aac379e59fa4c63350

Reply:
interesting is this using system six or if the virtualenv is used why not to switch to the recent six as we did for teuthology?

Reply:
It's using the system one.